console/Command.php

```diff
@@ -109,17 +116,23 @@ protected function _init() {
 		if (!is_object($this->request) || !$this->request->params) {
 			return;
 		}
-		$this->response = $this->_config['response'];
 
-		if (!is_object($this->response)) {
-			$this->response = $this->_instance('response', $this->response);
-		}
 		$default = array('command' => null, 'action' => null, 'args' => null);
 		$params = array_diff_key((array) $this->request->params, $default);
 
 		foreach ($params as $key => $param) {
+			if($key == 'no-color') {
+				$key = 'noColor';
+			}
+
 			$this->{$key} = $param;
 		}
+
+		$this->response = $this->_config['response'];
```

Comment:
This has been reordered, so that all request sections are dealt with before the response, this is because the response depends on the noColor section of the request.

The alternative is to process noColor first, and then process the rest afterwards.

This made more sense, and since none of the rest of the response depended on the request, it seems fine to do this way.